requirements/0004-volunteer--host-identity-validated.md

```diff
@@ -0,0 +1,175 @@
+
+When redirecting user to Authologic for identity verification, system SHALL provide Authologic with link to Verification Status View.
+
+##### ... I want to check my verification status
+
+System MUST allow user to check their verification status. If verification was successful system SHALL show a thank-you message to the user. If verification is Pending or Rejected system SHALL show a generic wait message.
+
+
+##### ... I want to recover my account
+
+When user tries to register with existing phone number system SHALL show Account Recovery View.
+Account Recovery View SHALL notify user that this option is not available and to call Crisis Point's Hot Line.
+
+### User Interface
+
+No impact.
+
+#### API
+
+New endpoints SHALL be created for this feature. These endpoints SHALL be available without authentication.
+
+### Services
+
+#### New Host Registration Service (POST /registration)
+
+If phone number sent in POST data already exists in database system SHALL return HTTP 302 and re-render generic Account Recovery View.
+
+If phone number sent in POST data does not exist in database, System:
+- SHALL create *Host* entry using data from POST, and set its status to Verification Pending.
+- SHALL create *AccommodationUnit* entry using data from POST, and set its status to Verification Pending.
+- SHALL create *VerificationSession* entry referencing newly created *Host* and *AccommodationUnit*.
+- SHALL use newly created *VerificationSession*'s GUID to form `callback` pointing at `POST /registration/{GUID}` endpoint.
+- SHALL use newly created *Host*'s GUID to form `returnUrl` pointing at Host Verification Status View.
+
+#### Host Identity Verification Service (POST /registration/{verificationSessionId})
+
+System SHALL only accept valid requests. Sender verification per [Authologic documentation](https://sandbox.authologic.com/docs/api/pl/v1.1/api/callback.html#weryfikacja-nadawcy) SHALL be determined as follows:
+- `X-Signature-Timestamp` SHALL be no more than 5 minutes different from server time when last byte of message was received.
+- `HMAC_SHA_256` hash of `<X-Signature-Timestamp>:<HTTP Response Body>` using secret signature key SHALL be equal to `X-Signature` header. When calculating the hash raw HTTP Body exactly as in Response SHALL be used, without any reformatting that might be introduced by backend frameworks.
+
+When valid response is received from Authologic, system SHALL examine `result.identity.status` field of the response. System SHALL set verification status of both *Host* and *AccommodationUnit* correlated with given *VerificationSession* to:
+- Verified if received status is `FINISHED` 
+- Rejected if received status is `FAILED`
+- Verification pending if received status is `IN_PROGRESS` or `PARTIAL`
+
+When *Host* verification status is changed to Verified, system SHALL send and email to the given email address notifying a *Host* of the fact.
```

Comment:
I think, talking to UX, that the flow would be such that the system let's host register in DB and send them verification link. Mainly, because per Filip's statement - there is a lot of registrations done over the phone, such that a volunteer would register the host, and would have "Send verification mail" button - and this way the process would be consistent.

And if so - we get email verification kind of for free.

Reply:
Authologic says they have not yet had clients send links via emails. There should be no technological hurdles with that, it’s just uncharted territory.